term_unix_ioctl.go

```diff
@@ -0,0 +1,58 @@
+// +build linux darwin freebsd openbsd netbsd dragonfly
+
+// Copyright 2013-2015 Bowery, Inc.
+// Copyright 2017 Attila Fülöp <attila@fueloep.org>
+
+package prompt
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+	"errors"
+)
+
+type Termios syscall.Termios
+
+// setTermios does the system dependent ioctl calls
+func getTermios(fd uintptr, req uintptr) (*Termios, error) {
+	termios := new(syscall.Termios)
+	_, _, err := syscall.Syscall(syscall.SYS_IOCTL, fd, req,
+		uintptr(unsafe.Pointer(termios)))
+
+	if err != 0 {
+		return nil, errors.New(err.Error())
```

Comment:
This `errors.New(err.Error())` is unnecessary (get error string to create a new error). Same thing in line35 and line55. But minor issue.